src/vis/interfaces.ts

```diff
@@ -169,6 +170,8 @@ export type PlotlyData = {
   xTickLabels?: string[];
   yTicks?: string[];
   yTickLabels?: string[];
+  xDomain?: [number, number];
```

Comment:
@dvzacharycutler the domain should support passing [0, undefined]. Then the typing should be [number | undefined, number | undefined].

Reply:
This is a bit tricky, since if we set these domains we have to set both numbers in the plotly layout. I can get [0, undefined] specifically to work, by using the clamp to 0 setting, but [1, undefined] wouldn't be possible without manually calculating maximum/minimum myself in Vis. That's probably not a huge deal though? Shouldn't happen frequently

Reply:
I guess it's a good thing to calculate min (if undefined) and/or max (if undefined). Should just be a few lines, and it won't be slower as plotly has to do it otherwise/it doesn't happen that often.